pandas/tests/base/test_unique.py

```diff
@@ -24,6 +24,9 @@ def test_unique(index_or_series_obj):
         expected = pd.MultiIndex.from_tuples(unique_values)
         expected.names = obj.names
         tm.assert_index_equal(result, expected)
+    elif isinstance(obj, pd.NumIndex):
+        expected = pd.NumIndex(unique_values, dtype=obj.dtype)
+        tm.assert_index_equal(result, expected)
```

Comment:
should we start using `exact=True` more aggressively?

Reply:
I think so too. `exact=True` should be the default IMO, though the function is public, so needs a deprecation process.

I can add `exact=True` to the occurances when I call this function (later).